src/crontab.c

```diff
@@ -479,16 +479,129 @@ static void delete_cmd(void) {
 		fprintf(stderr, "path too long\n");
 		exit(ERROR_EXIT);
 	}
+	
+	if (backup_crontab(n) == -1) {
+		fprintf(stderr, "no crontab for %s\n", User);
+		exit(ERROR_EXIT);
+	}
+
 	if (unlink(n) != 0) {
-		if (errno == ENOENT)
-			fprintf(stderr, "no crontab for %s\n", User);
-		else
-			perror(n);
+		perror(n);
 		exit(ERROR_EXIT);
 	}
 	poke_daemon();
 }
 
+/* returns	0	on success
+ *		-1	on non existent crontab file
+ *		-2	on failure to write the backup file
+ */
+static int backup_crontab(const char * crontab_path) {
+	const char *env_value;
+	char backup_dir[MAX_FNAME], backup_path[MAX_FNAME];
+	int ch = '\0';
+	FILE *crontab_file;
+	FILE *backup_file;
+	struct stat sb;
+
+	
+	/* create backup directory */
+	if ((env_value = getenv("XDG_CACHE_HOME")) != NULL) {
+		strcpy(backup_dir, env_value);
```

Comment:
Please use `glue_strings(backup_dir, sizeof backup_dir, env_value, "", '\0')` and check the return value to avoid buffer overflow.